src/test/java/com/google/sps/tests/ResultsServletTest.java

```diff
@@ -54,16 +72,35 @@ private String mockServlet(String keyValue) throws IOException {
   }
 
   @Test
-  public void testServlet() throws IOException {
-    String expected = "value";
-    String actual = mockServlet(expected);
+  public void testLoggedIn() throws IOException {
+    String key = "value";
+    String logoutURL = "url";
+    boolean isUserLoggedIn = true;
+    setUpUserServiceMock(isUserLoggedIn, logoutURL);
+    String actual = mockServlet(key);
+    String expected = String.format(LOGGED_IN_RESULT_TEMPLATE, key, logoutURL);
+    Assert.assertTrue(actual.contains(expected));
+  }
+
+  @Test
+  public void testLoggedOut() throws IOException {
+    String key = "value";
```

Comment:
[Optional] because you use key and logoutUrl in a few tests with these values, you can maybe declare them global constants.